flang/include/flang/Lower/ReductionRuntime.h

```diff
@@ -16,59 +16,92 @@ class ExtendedValue;
-  /// This calls the descriptor based runtime call implementation of the scan
-  /// intrinsics.
-  void genAllDescriptor(Fortran::lower::FirOpBuilder & builder,
-                        mlir::Location loc, mlir::Value resultBox,
-                        mlir::Value maskBox, mlir::Value dim);
-
-  /// Generate call to any runtime routine.
-  /// This calls the descriptor based runtime call implementation of the scan
-  /// intrinsics.
-  void genAnyDescriptor(Fortran::lower::FirOpBuilder & builder,
-                        mlir::Location loc, mlir::Value resultBox,
-                        mlir::Value maskBox, mlir::Value dim);
-
-  /// Generate call to all runtime routine. This version of all is specialized
-  /// for rank 1 mask arguments.
-  /// This calls the version that returns a scalar logical value.
-  mlir::Value genAll(Fortran::lower::FirOpBuilder & builder, mlir::Location loc,
-                     mlir::Value maskBox, mlir::Value dim);
-
-  /// Generate call to any runtime routine. This version of any is specialized
-  /// for rank 1 mask arguments.
-  /// This calls the version that returns a scalar logical value.
-  mlir::Value genAny(Fortran::lower::FirOpBuilder & builder, mlir::Location loc,
-                     mlir::Value maskBox, mlir::Value dim);
-
-  /// Generate call to Maxloc intrinsic runtime routine. This is the version
-  /// that does not take a dim argument.
-  void genMaxloc(Fortran::lower::FirOpBuilder & builder, mlir::Location loc,
-                 mlir::Value resultBox, mlir::Value arrayBox,
-                 mlir::Value maskBox, mlir::Value kind, mlir::Value back);
-
-  /// Generate call to Maxloc intrinsic runtime routine. This is the version
-  /// that takes a dim argument.
-  void genMaxlocDim(Fortran::lower::FirOpBuilder & builder, mlir::Location loc,
-                    mlir::Value resultBox, mlir::Value arrayBox,
-                    mlir::Value dim, mlir::Value maskBox, mlir::Value kind,
-                    mlir::Value back);
-
-  /// Generate call to Minloc intrinsic runtime routine. This is the version
-  /// that does not take a dim argument.
-  void genMinloc(Fortran::lower::FirOpBuilder & builder, mlir::Location loc,
-                 mlir::Value resultBox, mlir::Value arrayBox,
-                 mlir::Value maskBox, mlir::Value kind, mlir::Value back);
-
-  /// Generate call to Minloc intrinsic runtime routine. This is the version
-  /// that takes a dim argument.
-  void genMinlocDim(Fortran::lower::FirOpBuilder & builder, mlir::Location loc,
-                    mlir::Value resultBox, mlir::Value arrayBox,
-                    mlir::Value dim, mlir::Value maskBox, mlir::Value kind,
-                    mlir::Value back);
+class FirOpBuilder;
+
+/// Generate call to all runtime routine.
+/// This calls the descriptor based runtime call implementation of the scan
+/// intrinsics.
+void genAllDescriptor(Fortran::lower::FirOpBuilder &builder, mlir::Location loc,
```

Comment:
Reviewers: you can go to the gear symbol above and check the box to hide whitespace changes. That will clean out a lot of noise from the unindent changes.